### PR TITLE
EVM remove receipt type

### PR DESCRIPTION
**Pull request by bkolad**
# Description
This PR introduces the following changes:
1. Adds `BcsCodec` in `sov-state`. `bincode & postcard` have problems with serde `[serde(flatten)]` (https://github.com/bincode-org/bincode/issues/245) used by `etheres::Receipt` struct.
```
panicked at 'called `Result::unwrap()` on an `Err` value: SequenceMustHaveLength'
```
2. Replaces custom `Receipt` from the `EVM` module with `etheres::Receipt`. 

## Testing
Ci passes.

## Docs
No doc changes.

### CODE REVIEW

**Review by neysofu — COMMENTED** [review]
Just one comment about `codec` file organization :100:

File: module-system/sov-state/src/codec.rs
Comment (neysofu) [inline module-system/sov-state/src/codec.rs:1]: Three files for barely 200 LoCs seems like spreading it too thin. Do you think just one file is cluttered? as an alternative, maybe `codec/` is better and we won't pollute the root source directory as much.

Comment (bkolad) [reply module-system/sov-state/src/codec.rs:1]: I like that different codecs are in different files, the separation is very clear. I added `codec/`

Comment (neysofu) [reply module-system/sov-state/src/codec.rs:1]: 👍

**Review by neysofu — APPROVED** [review]

**Pull request merged**